### PR TITLE
docs: add install instructions for arch

### PR DESCRIPTION
**Pull request by FFY00**
Signed-off-by: Filipe Laíns <lains@archlinux.org>

### CODE REVIEW

**Comment by FFY00** [comment]
Yosys will always be at the latest version since we follow the upstream closely. Arch is a rolling distribution so there are no installation with repo access to an older yosys.

As a packager I'd like to advise to install nmigen via the system package manager, this is the only proper way to install it system-wide. Does it make sense to do it here?

**Comment by whitequark** [comment]
> Yosys will always be at the latest version since we follow the upstream closely. Arch is a rolling distribution so there are no installation with repo access to an older yosys.

Great!

> As a packager I'd like to advise to install nmigen via the system package manager, this is the only proper way to install it system-wide. Does it make sense to do it here?

Depends. Would Arch provide the latest release or the latest master?

**Comment by FFY00** [comment]
Latest release, arch packages stable software.

**Comment by whitequark** [comment]
Sure. Then you can add the Arch-specific instructions to the `Latest release` section, also with `:altname: linux`.

**Comment by whitequark** [comment]
Note that, in the future (as in, when I figure out the least footgun-y path towards it) the primary mode of using nMigen would be through per-project virtualenvs. Once that happens, these installation instructions would be primarily targeted at developers who a are just trying it out and don't want to bother with relatively heavyweight project management yet.

File: docs/install.rst [hunk 1]
```diff
@@ -68,6 +68,16 @@ Installing prerequisites
...
 
+   .. platform-choice:: arch
+      :altname: linux
+      :title: Arch Linux
+
+      Install GTKWave and Yosys by running:
+
+      .. code-block:: console
+
+         $ sudo pacman -S gtkwave yosys
```
Comment (whitequark) [inline docs/install.rst:79]: I think this should include Python (and pip) in case the reader wants to install an unreleased version of nMigen.

Comment (FFY00) [reply docs/install.rst:79]: Ah, right.

File: docs/install.rst [hunk 2]
```diff
@@ -130,6 +140,14 @@ Latest release
 
          $ pip3 install --user --upgrade nmigen
 
+   .. platform-choice:: arch
+      :altname: linux
+      :title: Arch Linux
+
+      .. code-block:: console
```
Comment (whitequark) [inline docs/install.rst:147]: Missing `|release:install|`

**Comment by whitequark** [comment]
Thanks!

**Pull request merged**